File: src/codility/prefix_sum/MinAvgTwoSlice.java
package codility.prefix_sum;

public class MinAvgTwoSlice {

    public int solution(int[] A) {
        int minIdx;
        double minAvg = (A[0] + A[1]) / 2D;
        minIdx = 0;

        for (int i = 2; i < A.length; i++) {
            double avg = (A[i - 1] + A[i]) / 2D;

            if (avg < minAvg) {
                minAvg = avg;
                minIdx = i - 1;
            }

            avg = (A[i - 2] + A[i - 1] + A[i]) / 3D;

            if (avg < minAvg) {
                minAvg = avg;
                minIdx = i - 2;
            }

        }
        return minIdx;
    }
}

File: src/codility/kotlin/MinAvgTwoSlice.kt
package codility.kotlin

class MinAvgTwoSlice {

    fun solution(A: IntArray): Int {
        // write your code in Kotlin

        var lowIndex = 0
        var lowSlice = (A[0] + A[1]).toDouble() / 2

        for (i in 2 until A.size) {
            val avg3 = getAvg(A[i - 2], A[i - 1], A[i])

            if (avg3 < lowSlice) {
                lowIndex = i - 2
                lowSlice = avg3
            }

            val avg2 = getAvg(A[i - 1], A[i])
            if (avg2 <

                    lowSlice) {
                lowIndex = i - 1
                lowSlice = avg2
            }

        }

        return lowIndex

    }

    private fun getAvg(vararg values: Int): Double {
        return values.sum().toDouble() / values.size.toDouble()
    }
}